build.rs

```diff
@@ -157,3 +171,125 @@ fn main() {
         }
     }
 }
+
+#[cfg(feature = "vendored")]
+fn make_zlib(compiler: &cc::Tool, src_dir: &path::PathBuf, out_dir: &path::PathBuf) {
+    use nix::fcntl;
+    use std::os::fd::AsRawFd;
+
+    // lock README such that if two crates are trying to compile
+    // this at the same time (eg libbpf-rs libbpf-cargo)
+    // they wont trample each other
```

Comment:
Have you actually seen this happen? I think Cargo will prevent this.

Reply:
I have not see issues compiling either zlib nor libbpf.

However, during development prior to using a feature flag, elfutils had a 100% reproducible problem when compiling a program which uses libbpf-rs as a dependency and libbpf-cargo as a build-dependency.